src/test_utils/katana/mod.rs

```diff
@@ -161,6 +165,54 @@ impl<'a> Katana {
         &self.sequencer
     }
 
+    /// Adds mock logs to the database.
+    pub async fn add_mock_logs(&self, n_logs: usize) {
+        // Get the Ethereum provider instance.
+        let provider = self.eth_provider();
+
+        // Get the database instance from the provider.
+        let database = provider.database();
+
+        // Create a mock log object with predefined values.
+        let log = Log {
+            inner: alloy_primitives::Log {
+                address: Address::with_last_byte(0x69),
+                data: alloy_primitives::LogData::new_unchecked(
+                    vec![B256::with_last_byte(0x69)],
+                    Bytes::from_static(&[0x69]),
+                ),
+            },
+            block_hash: Some(B256::with_last_byte(0x69)),
+            block_number: Some(0x69),
+            block_timestamp: None,
+            transaction_hash: Some(B256::with_last_byte(0x69)),
+            transaction_index: Some(0x69),
+            log_index: Some(0x69),
+            removed: false,
+        };
+
+        // Create a vector to hold all the BSON documents to be inserted
+        let mut log_docs: Vec<Document> = Vec::with_capacity(n_logs);
+
+        // Loop to create and collect the specified number of logs.
+        for _ in 0..n_logs {
+            // Create a StoredLog object from the predefined log.
+            let stored_log = StoredLog { log: log.clone() };
+            // Serialize the StoredLog object to a BSON document.
+            let bson_doc = bson::to_document(&stored_log).expect("Failed to serialize StoredLog to BSON");
+            // Add the BSON document to the vector.
+            log_docs.push(bson_doc);
+        }
```

Comment:
can't this also be replaced with a `into_iter().map()`?

Reply:
Yes a lot cleaner, just fixed